lib/puppet/application/device.rb

```diff
@@ -174,18 +178,17 @@ def main
     # find device list
     require 'puppet/util/network_device/config'
     devices = Puppet::Util::NetworkDevice::Config.devices
-    if options[:node]
-      devices.select! { |key, value| key == options[:node] }
+    if options[:target]
+      devices.select! { |key, value| key == options[:target] }
```

Comment:
The change from `node` to `target` lost the context on this, but here we go:

`devices` here is not a copy. It's a reference to the same devices hash that is stored in `Puppet::Utili::NetworkDevice::Config`. If you want a true copy, you'll want to use `.dup` as @MosesMendoza suggested, or just get rid of the `!` on the select.

My first comment presumed that you were intentionally trying to propagate the change through into the global state to force other parts of Puppet to use the filtered list. This was why I requested cleanups to avoid using global state in that way.

If we're just trying to filter locally, I'd go with a `dup` or a non-mutating `select`